packages/playground/remote/src/lib/worker-utils.ts

```diff
@@ -33,18 +33,23 @@ export type ParsedStartupOptions = {
 };
 
 export const receivedParams: ReceivedStartupOptions = {};
-const url = self?.location?.href;
-if (typeof url !== 'undefined') {
-	const params = new URL(self.location.href).searchParams;
-	receivedParams.wpVersion = params.get('wpVersion') || undefined;
-	receivedParams.phpVersion = params.get('phpVersion') || undefined;
-	receivedParams.storage = params.get('storage') || undefined;
-	// Default to CLI to support the WP-CLI Blueprint step
-	receivedParams.sapiName = params.get('sapiName') || 'cli';
-	receivedParams.phpExtensions = params.getAll('php-extension');
-	receivedParams.siteSlug = params.get('site-slug') || undefined;
-}
+self.addEventListener('message', (event) => {
+	if (event.data?.type === 'startup-options') {
+		const { startupOptions } = event.data;
+		receivedParams.wpVersion = startupOptions.wpVersion;
+		receivedParams.phpVersion = startupOptions.phpVersion;
+		receivedParams.sapiName = startupOptions.sapiName;
+		receivedParams.storage = startupOptions.storage;
+		receivedParams.phpExtensions = startupOptions.phpExtensions;
+		receivedParams.siteSlug = startupOptions.siteSlug;
+	}
+});
 
+// TODO find a better way to pass the startup options
+while (!receivedParams.wpVersion) {
+	// wait for the next event loop
+	await new Promise((resolve) => setTimeout(resolve, 0));
+}
```

Comment:
When the worker is loaded from the cache, query strings are in the URL but `self.location` doesn't contain the query string.

I tried modifying the response object, but it's read-only. 
Hash data works in the same way as query strings, so we can't switch to that.

The only solution I found is to pass `startupOptions` as a message to the worker.

The current implementation is a dirty draft solution, but it works. 
I'm now searching for a cleaner solution.